components/apps_form/apps_form_container.tsx

```diff
@@ -0,0 +1,229 @@
+            type: 'form',
+            values: {
+                name: field.name,
+                values,
+                value,
+            },
+        };
+
+        try {
+            const res = await this.props.actions.doAppCall(outCall);
+
+            if (res?.data?.form) {
+                this.setState({form: res.data.form});
+            }
+
+            return res;
+        } catch (e) {
+            return {data: {type: 'error', error: e.message}};
+        }
+    };
+
+    performLookupCall = async (field: AppField, values: AppFormValues, userInput: string): Promise<AppSelectOption[]> => {
+        if (!field.refresh) {
+            return [];
+        }
+
+        const call = this.getCall();
+        if (!call) {
+            return [];
+        }
+
+        const res = await this.props.actions.doAppCall({
+            ...call,
+            type: 'lookup',
+            values: {
+                user_input: userInput,
+                values,
+                name: field.name,
+
+                // form, // instead of including the form, just make sure any form elements that have a blank value in the `values` field here
+            },
+        });
+
+        if (res.data.type === AppCallResponseTypes.ERROR) {
+            return [];
+        }
+
+        const data = res.data.data as {items: AppSelectOption[]};
+        if (data.items && data.items.length) {
+            return data.items;
+        }
+
+        return [];
+    }
+
+    getCall = (): AppCall | null => {
+        const {postID, channelID, teamID} = this.props;
+        const {form} = this.state;
+
+        const propsCall = this.props.call;
+        const call = (form && form.call) || propsCall;
+
+        if (!call) {
+            return null;
+        }
+
+        return {
+            ...call,
+            context: {
+                app_id: call.context.app_id,
+                location: postID ? AppsBindings.IN_POST : call.context.location,
+                post_id: postID || call.context.post_id,
+                team_id: postID ? teamID : call.context.team_id,
+                channel_id: postID ? channelID : call.context.channel_id,
+            },
+        };
+    }
+
+    getContext = (): AppContext | null => {
+        const {call, postID, channelID, teamID} = this.props;
+
+        if (!call) {
+            return null;
+        }
+
+        return {
+            app_id: call.context.app_id,
+            location: postID ? AppsBindings.IN_POST : call.context.location,
+            post_id: postID,
+            team_id: postID ? teamID : call.context.team_id,
+            channel_id: postID ? channelID : call.context.channel_id,
+        };
+    }
+
+    onHide = () => {
+        this.props.onHide();
+    };
+
+    render() {
+        const {call} = this.props;
+        if (!call) {
+            return null;
+        }
+
+        const {form} = this.state;
+        if (!form) {
+            return null;
+        }
+
+        const dialogProps = {
```

Comment:
```suggestion
        const formProps = {
```

Reply:
If the plan is to rename dialog -> form, I would do a grep of dialog in the repo and rename found instances accordingly.

Reply:
And if we are moving from modal to form, I would also do a grep of `modal` and replace found instances